src/repart/repart.c

```diff
@@ -6041,21 +6041,23 @@ static int append_btrfs_subvols(char ***l, OrderedHashmap *subvolumes, const cha
         assert(l);
 
         ORDERED_HASHMAP_FOREACH(subvolume, subvolumes) {
-                _cleanup_free_ char *s = NULL, *f = NULL;
+                _cleanup_free_ char *s = NULL, *f = NULL, *p = NULL;
 
-                s = strdup(subvolume->path);
-                if (!s)
+                p = strdup(subvolume->path);
+                if (!p)
                         return log_oom();
 
                 f = subvolume_flags_to_string(subvolume->flags);
                 if (!f)
                         return log_oom();
 
-                if (streq_ptr(subvolume->path, default_subvolume) &&
-                    !strextend_with_separator(&f, ",", "default"))
+                if (streq_ptr(subvolume->path, default_subvolume) && !strextend(&s, "default"))
+                        return log_oom();
+
+                if (!isempty(f) && !strextend_with_separator(&s, "-", f))
```

Comment:
Currently, only ro or default-ro will be set, so this should work.
But, in general the logic is incorrect, as subvolume_flags_to_string() uses strv_join() and potentially concat multiple options with comma, but btrfs command taks only one option.

So, please also update subvolume_flags_to_string().

Reply:
I'm tempted to remove `subvolume_flags_to_string` (as in latest push). Inlining the RO test is simple, and the new API doesn't look like it will be extended in that way. This is the only use of the function.

@DaanDeMeyer, do you expect we'll need this helper for future changes in btrfs-progs, or are you happy with us dropping `subvolume_flags_to_string`?

Reply:
Please drop it